flows/parent_flow.py

```diff
@@ -1,47 +1,367 @@
+    if hpc_type == "nersc":
+        logger.info(f"HPC type is NERSC, selecting SLURM")
+        return FlowType.slurm
+    elif hpc_type == "nsls-ii":
+        logger.info(f"HPC type is NSLS-II, selecting PODMAN")
+        return FlowType.podman
+    elif hpc_type == "als":
+        logger.info(f"HPC type is ALS cluster-ball, selecting DOCKER")
+        return FlowType.docker
+    elif hpc_type in [ft.value for ft in FlowType]:
+        # If the hpc_type is actually a flow type, use it directly
+        return FlowType(hpc_type)
+    else:
+        # Default to conda for unknown HPC types
+        logger.info(f"Unknown HPC type: {hpc_type}, defaulting to CONDA environment")
+        return FlowType.conda
+
+@task
+def get_algorithm_details_from_mlflow(model_name: str):
+    """
+    Retrieve algorithm details from MLflow using the model name.
+    
+    Args:
+        model_name: The name of the model in MLflow
+    
+    Returns:
+        Dictionary containing algorithm details
+    """
+    logger = get_run_logger()
+    logger.info(f"Retrieving details for model {model_name} from MLflow")
+    
+    # Log MLflow connection parameters for debugging
+    logger.info(f"MLflow Tracking URI: {MLFLOW_TRACKING_URI}")
+    logger.info(f"MLflow Username: {'Set' if MLFLOW_TRACKING_USERNAME else 'Not set'}")
+    logger.info(f"MLflow Password: {'Set' if MLFLOW_TRACKING_PASSWORD else 'Not set'}")
+    
+    # Set MLflow connection
+    os.environ["MLFLOW_TRACKING_USERNAME"] = MLFLOW_TRACKING_USERNAME
+    os.environ["MLFLOW_TRACKING_PASSWORD"] = MLFLOW_TRACKING_PASSWORD
+    mlflow.set_tracking_uri(MLFLOW_TRACKING_URI)
+    
+    try:
+        client = MlflowClient()
+        
+        # Add additional logging to list all registered models
+        try:
+            all_models = client.search_registered_models()
+            logger.info(f"Found {len(all_models)} registered models in MLflow:")
+            for rm in all_models:
+                logger.info(f"  - {rm.name}")
+        except Exception as e:
+            logger.warning(f"Could not list registered models: {str(e)}")
+        
+        # Get the latest version of the model
+        logger.info(f"Attempting to get latest versions for model: {model_name}")
+        versions = client.get_latest_versions(model_name)
+        if not versions:
+            logger.error(f"No versions found for model {model_name}")
+            raise ValueError(f"Model {model_name} not found in MLflow")
+            
+        version = versions[0]
+        logger.info(f"Found version {version.version} for model {model_name}")
+        
+        # Get the run to access parameters
+        run = client.get_run(version.run_id)
+        logger.info(f"Retrieved run with ID: {run.info.run_id}")
+        
+        # Extract the relevant parameters
+        params = run.data.params
+        tags = run.data.tags
+        
+        # Get relevant fields from MLflow params
+        algorithm_details = {
```

Comment:
I’d recommend simplifying this section to focus on metadata that’s directly relevant to the algorithm itself, rather than environment or runtime configuration (e.g. `conda_env`, `network`, `volumes`, etc.):

```suggestion
        algorithm_details = {
            "model_name": model_name,
            "image_name": params.get("image_name", ""),
            "image_tag": params.get("image_tag", ""),
            "source_code": params.get("source_code", ""),  # e.g. repo link 
        }
```

Reply:
Thanks for your suggestion. I change `algorithm_details` to
```
  algorithm_details = {
      "model_name": model_name,
      # Core Algorithm Information
      "image_name": params.get("image_name", ""),
      "image_tag": params.get("image_tag", ""),
      "source": params.get("source", ""),
      "is_gpu_enabled": params.get("is_gpu_enabled", "False").lower() == "true"
  }
```
And create another `job_details` to load information from the newly created `config.yml` file:
```
job_details = {
    # Container settings
    "volumes": config.get("container", {}).get("volumes", []),
    "network": config.get("container", {}).get("network", ""),
    # Slurm settings
    "num_nodes": config.get("slurm", {}).get("num_nodes", 1),
    "partitions": config.get("slurm", {}).get("partitions", "[]"),
    "reservations": config.get("slurm", {}).get("reservations", "[]"),
    "max_time": config.get("slurm", {}).get("max_time", "1:00:00"),
    "submission_ssh_key": config.get("slurm", {}).get("submission_ssh_key", ""),
    "forward_ports": config.get("slurm", {}).get("forward_ports", "[]"),
    # Get conda environment based on the model type
    "conda_env": _get_conda_env_for_model(model_name, config)
}
```